### ISSUE TITLE
replacing storybook with styleguidist

### ISSUE THREAD

**Issue by Introvertuous** [airbnb/react-dates#1206]
Not sure if you guys have considered this before but that demo page does look great. Styleguidist is more appropriate for showcasing components, describing prop types, etc, whearas storybook works better as a dev tool.

ex.
https://bokuweb.github.io/re-bulma/

**Comment by majapw** [comment]
Styleguidist is very cool! I guess I like the fact that storybook is *both* a showcase and a dev-tool simultaneously? I'm not sure how much benefit we'd get from both/how much the cost of supporting both would be.

**Comment by Introvertuous** [comment]
Yep, they both have areas in which they excel. I would be willing to do up a PR for styleguidist support if you wanted to see the workflow. I find it really nice, you essentially just document your components with markdown.

**Comment by majapw** [comment]
If it's not a ton of trouble, I'd love to play with a prototype. :)

**Comment by Introvertuous** [comment]
Sure thing, I will do that on the weekend.

**Comment by Introvertuous** [comment]
Initial PR: #1230
Only documented one component (DateRangePicker).
Still needs work but should give you the idea.
